frontend/app/routes/api/session.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * An API route that can be used to perform actions with user's server-side session.
+ */
+import type { ActionFunctionArgs } from '@remix-run/node';
+import { json } from '@remix-run/node';
+
+import { z } from 'zod';
+
+import { getSessionService } from '~/services/session-service.server';
+import { getLogger } from '~/utils/logging.server';
+
+const log = getLogger('routes/api/session');
+
+export enum ApiSessionAction {
+  End = 'end',
+  Extend = 'extend',
+}
+
+const bodySchema = z.object({
+  action: z.nativeEnum(ApiSessionAction),
+});
+
+export async function action({ context: { session }, request }: ActionFunctionArgs) {
+  log.debug("Action with user's server-side session");
+
+  if (request.method !== 'POST') {
+    log.warn('Invalid method requested [%s]; responding with 405', request.method);
+    throw json({ message: 'Method not allowed' }, { status: 405 });
+  }
+
+  const requestBody = await request.json();
+  const parsedBody = bodySchema.safeParse(requestBody);
+
+  if (!parsedBody.success) {
+    log.debug('Invalid request body [%j]', requestBody);
+    return json({ errors: parsedBody.error.flatten().fieldErrors }, { status: 400 });
+  }
+
+  switch (parsedBody.data.action) {
+    case ApiSessionAction.End: {
+      log.debug("Ending user's server-side session");
+      const sessionService = await getSessionService();
+      const headers = { 'Set-Cookie': await sessionService.destroySession(session) };
+      return new Response(null, { headers, status: 204 });
+    }
+
+    case ApiSessionAction.Extend: {
+      log.debug("Extending user's server-side session lifetime");
```

Comment:
Same comment as the one I made in `apply-state.ts` regarding logging the `id` of the `session`, which can also be applied to the `end` action debug log above.

Reply:
Fixed